### ISSUE TITLE
changes to default applications are not saved correctly or reset

### ISSUE THREAD

**Issue by ArjenR** [pop-os/cosmic-settings#808]
So far I have not determined what is happening to the fullest extents but it looks like the following is happening.

1. Change the default browser
2. Leave the screen
3. Re enter the default applications screen. Browser is like it was set in step 1.
4. Change the default email application
5. Leave and re-enter the screen
6. Browser is reset to state it was before step 1
7. Email client is as set in 4.
8. Change browser again.
9. Leave and enter default applications screen.
10. Email client is reset.

I think actually changing 1 item resets all others. The setting for the  music player is never saved. 
This system is an upgrade from earlier Pop version.

```
cosmic-settings/noble,now 0.1.0~1733335175~24.04~d01e0d8 amd64
cosmic-settings-daemon/noble,now 0.1.0~1732290079~24.04~f079ab7 amd64
```
edit: formatting

**Comment by QuantenToast** [comment]
I seem to have a related issue:
Changing anything in the default applications menu resets the `~/.config/mimeapps.list`, which, of course, leads to my other mime/application associations disappearing (e.g., images are now opening in Firefox instead of Gwenview).

cosmic-settings and cosmic-settings-deamon are version 1.0.0.alpha.4-1 (Arch Linux extra repository)

**Comment by davxy** [comment]
Also would be nice to have something like this: https://github.com/pop-os/cosmic-settings/issues/444#issuecomment-2323054470

**Comment by mmstick** [comment]
Fixed by 462a917f37b6f407cfdf5b9c907c71567e77b9ad
